### PR TITLE
vendor: bump github.com/evanphx/json-patch

### PR DESCRIPTION
**Pull request by philips**
Grab important bug fix that can cause a `panic()` from this package on
certain inputs. See https://github.com/evanphx/json-patch/commit/73af7f547e13ad70d7f7b3388255e7e5eb93adaf

```release-note
NONE
```

### CODE REVIEW

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -1522,8 +1522,8 @@
 		},
 		{
 			"ImportPath": "github.com/evanphx/json-patch",
-			"Comment": "v3.0.0-23-g94e38aa",
```
Comment (liggitt) [inline Godeps/Godeps.json]: I'm surprised none of the files under the staging directories needed this update as well

**Comment by philips** [comment]
Gah, you are right @liggitt.

/hold

**Comment by philips** [comment]
/hold cancel

@liggitt Ran the `./hack/update-staging-godeps.sh` script. Is there anything else I am missing? Sorry, it has been awhile since I have bumped a dep.

**Review by cblecker — APPROVED** [review]
/lgtm
/approve

**Review by dougm — COMMENTED** [review]
/lgtm

**Comment by dougm** [comment]
Merged to release-1.12 branch via branchff: https://github.com/kubernetes/kubernetes/commits/release-1.12

**Pull request merged**